framework/src/utils/MeshBaseDiagnosticsUtils.C

```diff
@@ -66,4 +66,107 @@ checkNonConformalMesh(const std::unique_ptr<MeshBase> & mesh,
   }
   pl->unset_close_to_point_tol();
 }
+
+bool
+checkEdgeOverlap(const std::unique_ptr<Elem> & edge1,
+                 const std::unique_ptr<Elem> & edge2,
+                 const ConsoleStream & console,
+                 const Real intersection_tol)
+{
+  // get node array from two edges
+  const auto node_list1 = edge1->get_nodes();
+  const auto node_list2 = edge2->get_nodes();
+
+  // make sure two edges are not the same and don't share any nodes edge before checking for overlap
+  auto n1 = *node_list1[0];
+  auto n2 = *node_list1[1];
+  auto n3 = *node_list2[0];
+  auto n4 = *node_list2[1];
+
+  // get x,y,z coordinates for each point
+  auto n1x = n1.operator()(0);
+  auto n1y = n1.operator()(1);
+  auto n1z = n1.operator()(2);
+  auto n2x = n2.operator()(0);
+  auto n2y = n2.operator()(1);
+  auto n2z = n2.operator()(2);
+  auto n3x = n3.operator()(0);
+  auto n3y = n3.operator()(1);
+  auto n3z = n3.operator()(2);
+  auto n4x = n4.operator()(0);
+  auto n4y = n4.operator()(1);
+  auto n4z = n4.operator()(2);
+
+  // Check that none of these points are the same
+  if (((std::fabs(n1x - n3x) < intersection_tol) && (std::fabs(n1y - n3y) < intersection_tol) &&
+       (std::fabs(n1z - n3z) < intersection_tol)) ||
+      ((std::fabs(n2x - n4x) < intersection_tol) && (std::fabs(n2y - n4y) < intersection_tol) &&
+       (std::fabs(n2z - n4z) < intersection_tol)) ||
+      ((std::fabs(n1x - n4x) < intersection_tol) && (std::fabs(n1y - n4y) < intersection_tol) &&
+       (std::fabs(n1z - n4z) < intersection_tol)) ||
+      ((std::fabs(n2x - n3x) < intersection_tol) && (std::fabs(n2y - n3y) < intersection_tol) &&
+       (std::fabs(n2z - n3z) < intersection_tol)))
```

Comment:
if the nodes overlap and are not the same

Reply:
So the reason I added that check was because I figured that if two nodes had the same x,y,z coordinates then they were the same node. I did try just checking if n1==n3 or n1==n4, ect. But there were a couple of cases where it didn't recognize the two nodes as equal, but comparing their coordinates did work.

Reply:
not necessarily in a broken mesh. we could have nodes that overlap but do not have the same id
checking the id is enough to check equality

Reply:
So I think I figured out why using n1==n3 wasn't working before. I had a bug where it was assigning the same id to every node. I did notice though that if it's implemented this way that with unmerged meshes with multiple volumes it will count all touching edges as intersecting since their nodes have different ids. I don't think that this is necessarily a bad thing but I wanted to double check that that's how we want it to behave?